pkg/utils/mime.go

```diff
@@ -0,0 +1,118 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
```

Comment:
It wasn't clear to me what the utility of this package was. Is this giving us some novel code that we need in order to parse the AL2023 userData, duplicating code from another package or something else?

Reply:
Mentioned in a previous comment but I factored this out for use in the launchtemplate suite test. Additionally, there is duplicated logic between the eksbootstrap and nodeadm, I'll probably move eksbootstrap over to this too.

Reply:
The functions in here might also be more idiomatic if the package name was `mime` vs. `utils`. e.g. `mime.Entry`, `mime.Archive`, etc.